### DISCUSSION TITLE
Improving the copyright line handling in the footer.php

### DISCUSSION THREAD

**Discussion by kreincke** [bootscore/bootscore#389] [Ideas]
The four footers can be filled by widgets. The copyright line at the bottom is a 'hard-coded' element of footer.php. What, if we integrate a function in functions.php by which the copyright line can be computed without adding the footer.php into the child-theme?

Advantage: If the child-theme developer has to upgrade bootScore it would be helpful if he had to merge as fewer files as possible.

If you agree with this proposal, I will create a respective pull request.

(BTW: When will the version be released? And which pr's will be taken over?)
heartfully KR

**Comment by kreincke** [comment]
After having studied your structure, I would now propose

- add `register_sidbar [...] crline` in functions.php (just as the elements are defined)
- replace the hardcoded cr-line in footer.php by `... is_active_sidebar(crline)....dynamic_sidebar(crline)...` (just as the other elements are evaluated.)

If you agree, I will create the respective pull request

Advantage:
- The bootscore user (theme editor) must design his CR Line in accordance to his needs.
- Yes, you've said that configuring bootscore has to be done (also) via programming. But by using such a function, you would ease to upgrade bootscore.

Disadvantage: you can not enforce your users NOT add a picture or anything else. But sometimes it is better to describe the restrictions and the intended usage instead of making 'abuse' more difficult

Best regards KR

**Comment by crftwrk** [comment]
Do you mean this `<small>&copy;&nbsp;<?php...`? What's wrong here?

Next release is scheduled Feb 15. You can track this in the projects board https://github.com/orgs/bootscore/projects/14.

**Comment by kreincke** [comment]
The content of the footers can be edited on Gutenberg level via a widget. Why shall we enforce the users to do that on file level for the copyright line (footer.php)? If we used a widget also for the cr-line, the user would not have to copy footer.php into his child theme. And the fewer files he must copy into his child theme, the easier is an update of bootscore-main, because for each existing copies he has to port his modifications in the updated files. For me, that seems to be reasonable.

Best regards KR

**Comment by crftwrk** [comment]
You mean to replace the copyright by another widget and user should do copyright manually in backend, right? In this case, copyright is just a new widget position but `footer.php` has already 5 widgets which is more than enough. 

BTW, this is not the spirit of a copyright because now you have an automated year and site title on your site and makes sense as it is. If you want to register more widgets in your footer, you can easily register them in your child. 

> the user would not have to copy footer.php into his child theme

But this is exactly what a user should do. A child-theme is there to copy files from main-theme and edit them in an upgrade-safe way. `footer.php` is always the first file I override in a new project.

We do not change this, it is good as it is.

**Comment by kreincke** [comment]
sorry for having bothered you. Won't happen again. bg kr

**Reply by crftwrk** [reply]
All good, we are here to discuss and improve things ;-)

**Comment by kreincke** [comment]
Please - if and when you have time - have a look at [https://fodina.de/2023/footers-c/](https://fodina.de/2023/footers-c/) and tell me if you feel uncomfortable with my summary of our discussion.
br kr

**Comment by crftwrk** [comment]
> tell me if you feel uncomfortable with my summary of our discussion.

No, all good. This is your opinion and have to be respected.

Let me try to explain again why we do not replace copyright by widget:

- Users who not have overridden footer in child has to update their copyright manually otherwise they will not have one anymore. We do not updates which breaks child.
- I personally have to manage round about 40 bootScore based sites for clients. I'm happy about the self-updated year because otherwise I have to update year on each page manually every January 1 again.
- bootScore provides a simple, pixel-perfect Bootstrap layout with a lot of functionality which fits for everyone and should be neutral as possible. Copyright as a widget is something which might be interesting for you but not for everyone.
- bootScore is a coding project and designed to edit files in child theme. Adding design and functionality to create awesome sites, is the users job. Theme provides all tools to be creative.

Hope that clears it up.

**Comment by kreincke** [comment]
Dear Basti(an), 

Please let me emphasize via this answer, that I appreciate your work very well. It was just what I was looking for when I had to reorganize my site . I do not want to give the impression that I am fighting with you. On the opposite, you are easing my life. Great job. Great idea. Many thanks 

Best regards Karsten

**Comment by crftwrk** [comment]
Thought about it again for a while. As mentioned in previous comment, we will not remove copyright for several reasons. 

But, I agree that we can add a new widget widget position above the copyright and give it a neutral name like `footer-info` to make users clear that this widget has nothing to do with a copyright. It's just a simple widget where you can show whatever you want. We can add a class to existing copyright `<small>`, for example `<small class="footer-copyright">` and users can hide this by CSS if they want. This should be exactly the result as you want.

Agree?

**Comment by muddg** [comment]
I think it would be a great improvement.

Thank you

**Comment by crftwrk** [comment]
https://github.com/bootscore/bootscore/pull/449

**Comment by kreincke** [comment]
I've now tested release 5.2.3.4 which offers both, the old hard-coded copyright line and the new footer info line - both presented at the bottom below a separating horizontal line. In accordance with the developers, the hard-coded cr-line can simply be suppressed by a CSS configuration. (cf. [https://fodina.de/footers-c/](https://fodina.de/footers-c/))

That's an excellent solution! 

heartfully KR
